Orange/widgets/settings.py

```diff
@@ -398,14 +398,17 @@ def write_defaults(self):
         filename = self._get_settings_filename()
         os.makedirs(os.path.dirname(filename), exist_ok=True)
 
-        settings_file = open(filename, "wb")
         try:
-            self.write_defaults_file(settings_file)
-        except (EOFError, IOError, pickle.PicklingError):
-            settings_file.close()
-            os.remove(filename)
-        else:
-            settings_file.close()
+            settings_file = open(filename, "wb")
+            try:
+                self.write_defaults_file(settings_file)
+            except (EOFError, IOError, pickle.PicklingError): # pragma: no cover
+                settings_file.close()
+                os.remove(filename)
+            else:
+                settings_file.close()
+        except PermissionError: # pragma: no cover
+            pass
```

Comment:
@astaric, I suppose that we should warn the user that settings were not saved, using a critical warning dialog? We'd do it just once per session.

This also goes for the exceptions caught above.